### PR TITLE
MGDOBR-441: Shard may never delete resources if it crashes after sending DELETING status call

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-441](https://issues.redhat.com/browse/MGDOBR-441)

I'm not entirely convinced this is the best approach and would welcome thoughts.

At the moment we have a small window of potential failure between us advising the `manager` that a resource is being provisioned and us actually asking k8s to create or replace the CRD. Once we've told k8s then we're bullet proof. I was toying with the idea of a much larger refactor that puts the "tell the manager" call and "tell k8s" much closer together; but even if they're subsequent lines there will always remain potential for failure.

What's in this PR works; but I am not convinced it adds more smell to the code than it fixes. 

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ManagerSyncServiceImpl.java [hunk 1]
```diff
@@ -81,6 +87,10 @@ protected Uni<Object> doBridges() {
                                         },
                                         failure -> failedToSendUpdateToManager(y, failure));
                     }
+                    if (y.getStatus().equals(ManagedResourceStatus.DELETING)) { // Bridges that were being deleted (before the Operator failed).
+                        LOGGER.info("Found Bridge '{}' in DELETING state. Attempting to continue with DELETING.", y.getId());
+                        return Uni.createFrom().voidItem().subscribe().with(success -> deleteBridgeIngress(y));
```
Comment (r00ta) [inline shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ManagerSyncServiceImpl.java:92]: is there any specific reason for `Uni.createFrom().voidItem().subscribe()...`? 
Actually in this case you can 
```java
deleteBridgeIngress(y);
return Uni.createFrom().voidItem();
```

same for the other cases

**Review by r00ta — COMMENTED** [review]
Looks good overall

File: shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ManagerSyncServiceImpl.java (hunk 1 above)
Comment (manstis) [reply shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ManagerSyncServiceImpl.java:92]: Yes, `Uni.createFrom().voidItem().invoke(()->...)` didn't actually run. I could only get the code to run by subscribing to it.

**Comment by manstis** [comment]
Putting up for offical review since @r00ta [did not assassinate the approach](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/946#pullrequestreview-1039396452) :smiley:

File: shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ManagerSyncServiceImpl.java (hunk 1 above)
Comment (r00ta) [reply shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ManagerSyncServiceImpl.java:92]: if you use 
```
deleteBridgeIngress(y);
return Uni.createFrom().voidItem();
```
it should work, can you try?

Comment (manstis) [reply shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ManagerSyncServiceImpl.java:92]: @r00ta that worked too. I'll make the change.

**Review by r00ta — APPROVED** [review]
approving as this PR is not assassinating the codebase like [this](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/929#discussion_r919690308) ;)

**Review by sutaakar — APPROVED** [review]
Looks good to me

**Comment by manstis** [comment]
@robpblake Both @r00ta and @sutaakar have approved (the "2 approvals" I need to merge).

Please advise if you'd like an opportunity to review; otherwise I will proceed to merge.

**Review by robpblake — APPROVED** [review]

**Pull request merged**